### PR TITLE
Remove the global wc-cart-fragments enqueue

### PR DESCRIPTION
**Pull request by mikejolley**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

`wc-cart-fragments` allows statically rendered elements to be updated when a product is added to the cart. It does this with an additional AJAX call to the `get_refreshed_fragments` endpoint. This does create a performance issue where extra requests are made to the server to support it.

There are several articles on the web detailing how this is a problem and recommending to dequeue it [1], as well as workarounds which replace the functionality entirely [2].

Given that the main consumer of this functionality is the **Cart Widget**, and that the Cart Widget is being superseded by the **Mini Cart Block**, I propose we remove this from the global enqueues and move it to the Cart Widget only, so the widget can continue to use fragments, but other pages in the store do not need to load the script.

This was proposed here: https://github.com/woocommerce/woocommerce-blocks/issues/7168

We plan on removing support for fragments in WooCommerce Blocks, and potentially dequeuing it completely in the interim.

- [1] https://wpengine.com/support/how-to-disable-cart-fragments-for-woocommerce/
- [2] https://wpengine.com/blog/solving-cart-fragment-dilemma-introducing-live-cart/

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. With the patch applied, and with no widgets setup, visit the shop page.
2. View the page source and confirm the `cart-fragments.js` script is not loaded.
3. Add the cart widget to a sidebar in the theme. 
4. View the page source and confirm the `cart-fragments.js` script is loaded.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/changelog/update-cart-fragments-enqueue [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: performance
+
+Removed global enqueue of wc-cart-fragments. Moved to the cart widget (it's main consumer).
```
Comment (beaulebens) [inline plugins/woocommerce/changelog/update-cart-fragments-enqueue]: Grammar: should be "its" in this case, not "it's" :)

Comment (Konamiman) [inline plugins/woocommerce/changelog/update-cart-fragments-enqueue]: Adding as suggestion comment to be able to do the change easily:

```suggestion
Removed global enqueue of wc-cart-fragments. Moved to the cart widget (its main consumer).
```

**Review by Konamiman — APPROVED** [review]

**Comment by mikejolley** [comment]
Ill go ahead and merge this before the feature freeze. The following note can be included with the release:

> The Cart Fragments script, responsible for updating the cart widget after a successful add to cart event, can cause a performance issue on stores not making use of that functionality. To mitigate this, the script will now only be enqueued if using the Mini Cart widget.
>
> Stores with similar functionality (e.g. from a theme) that are _not_ using the widget can re-enable this functionality by either adding the widget to one of their widget areas/sidebars, or by enqueuing the script manually:
> 
> ```wp_enqueue_script( 'wc-cart-fragments' );```
> 
> If you're looking for an alternative to the Mini Cart widget that does not suffer from the performance issue, try the **Mini Cart Block** instead.

**Comment by MehbubRashid** [comment]
Isn't it a breaking change? In my theme the code `$(document.body).trigger('wc_fragment_refresh')` is used in some places. After updating WooCommerce, I am noticing few features are not working in the frontend and at last i found the reason to be this change. Good for me that I noticed it. Other users who have the same code related to wc_fragment_refresh, they wont even know that the old features of their site is not working after updating WooCommerce.

**Pull request merged**